### PR TITLE
Make orchantiforgery tenants cookies secure

### PR DESCRIPTION
**Pull request by Skrypt**
Related to #15337

image

### CODE REVIEW

**Comment by sebastienros** [comment]
I believe it fails the integration tests because:

```
Secure is always marked true. Use this value when your login page and all subsequent pages requiring the authenticated identity are HTTPS. Local development will also need to be done with HTTPS urls.
```

So this will break if we run in http with this option.

I think this should be set only when https is made mandatory, so not by default, but maybe could be done in the Security feature when https is forced.

**Comment by Skrypt** [comment]
hmm, ok will rethink this one.

**Comment by Skrypt** [comment]
Ok, so our integrations test are executed with an environment var that sets it to production.

**Comment by sebastienros** [comment]
> The issue is that the functional tests

It's not just about functional tests. It's also about local dev experience. If you do this change then we need to start our default apps/templates on https only too.

File: src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -412,47 +468,64 @@ private static void AddAntiForgery(OrchardCoreBuilder builder)
...
-                // Re-register the antiforgery services to be tenant-aware.
-                var collection = new ServiceCollection()
-                    .AddAntiforgery(options =>
+                    // Re-register the antiforgery services to be tenant-aware.
+                    var collection = new ServiceCollection().AddAntiforgery(options =>
                     {
                         options.Cookie.Name = cookieName;
+                        options.Cookie.SecurePolicy = env.IsProduction()
+                            ? CookieSecurePolicy.Always
+                            : CookieSecurePolicy.SameAsRequest;
```
[the remark this answers is not in the document]
Comment (sebastienros) [reply src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs:484]: You can't do that: assume that production means it's https. You can only do that from the Secure feature that enforces https.

[the remark this answers is not in the document]
Comment (Skrypt) [reply src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs:484]: Ok, you are right, there are still people that uses http on production. Then we need to do what you are suggesting. When you say "secure feature" you mean the "Https feature" that we have? 

We can't read site settings from this ServiceCollectionExtension as the service is not yet registered.

[the remark this answers is not in the document]
Comment (sebastienros) [reply src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs:484]: In the https feature we can call AddAntiforgey and only set the secure property. This is cumulative.

**Comment by Skrypt** [comment]
Although I made the changes suggested here on this PR it doesn't seem to work at all. Even after clearing the cookies and restarting the app.

**Pull request closed without merging**